### DISCUSSION TITLE
Create an Instance of SB without context

### DISCUSSION THREAD

**Discussion by pol4xer** [seleniumbase/SeleniumBase#2595] [Q&A] [answered]
I am trying to use SB, but I don't like context:
```python3 
from seleniumbase import SB
with SB(...) as sb:
    pass
```

Is there any way to create an instance of SB and save it to a variable? Something like that:
```python3 
from seleniumbase import SB
class Foo(SB):
    pass

foo = Foo()
foo.uc_open(...)
foo.get_new_driver()
foo.open_new_tab()
...
```

Of course, I can use Driver: `from seleniumbase import Driver`, but it doesn't have `get_new_driver` and `open_new_tab` features.

**Comment by mdmintz** [comment, marked as answer]
For the `SB()` / `BaseCase` formats (seen in [help_docs/syntax_formats.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md)), there must always be a beginning AND an end. That way everything can be properly set up in the beginning, and properly closed at the end. Without a context manager format, there's no way of knowing when the end is reached. When used for website testing, the end part also includes code for generating test reports, etc.

For just the variable/object with no end code, there's the `Driver()` format, which has limited functionality. The `Driver()` object expects you to close the browser on your own via `driver.quit()` (to prevent memory leaks). The `Driver()` object also expects you to handle virtual display setup and teardown on your own (using https://github.com/mdmintz/sbVirtualDisplay) when you need to use SeleniumBase in a headless environment.

**Reply by pol4xer** [reply]
As I understood, SB is more "antibot" than pure "Driver", that's why I want to use SB. But context limits me in the process of developing. 
About beginning - I can setup everything in the __init__ of my custom class or do nothing, if I expect base logic from it.
About closing - I was planning to close browser after i'm done with it:
```python3
from seleniumbase import SB
class Foo(SB):
    pass

foo = Foo()
foo.uc_open(...)
foo.get_new_driver()
foo.open_new_tab()
...
foo.close() / foo.quit() / foo.exit() # or something like that
```

**Reply by mdmintz** [reply]
`SB()` is about being able to run SeleniumBase scripts with the full API without using `pytest`, which is what most scripts are run with (`BaseCase` formats). The UC Mode part is one of many options (`--uc` / `uc=True`).
